pyproject.toml

```diff
@@ -44,7 +44,7 @@ testpaths = [
 ]
 
 [tool.mypy]
-python_version = "3.11"
```

Comment:
do we want to keep this to 3.11 so we don't use any 3.12 specific syntax? I think for the static analysis you want to keep it to the minimum supported version